controllers/deployment_controller.go

```diff
@@ -46,12 +51,17 @@ func (r *DeploymentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 	deployment := new(csibaremetalv1.Deployment)
 	err := r.Get(ctx, client.ObjectKey{Name: req.Name, Namespace: req.Namespace}, deployment)
 	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Custom resource is not found")
```

Comment:
Should we implement logic about removing CSI deployment and DaemonSet, if CR has been deleted?

Reply:
No. It works automatically with OwnerReferences. When CR has DeletionTimestamp, all resources, which it controls, get DeletionTimestamp too
I will add finalizers and remove logic in the next issue :)